config/src/config/network_config.rs

```diff
@@ -191,6 +191,43 @@ impl NetworkConfig {
         };
         self.network_keypairs = Some(network_keypairs);
     }
+
+    /// Use this node config to build a seed peer address (using the given
+    /// `bootstrap_address` instead of the configured `self.advertised_address`).
+    pub fn build_seed_address(&self, bootstrap_address: NetworkAddress) -> Result<NetworkAddress> {
+        // TODO(philiphayes): use enum in wire crate when that gets extracted
+        let handshake_version = 0;
+
+        if self.enable_remote_authentication || self.enable_noise {
+            let pubkey = self
+                .network_keypairs
+                .as_ref()
+                .ok_or_else(|| {
+                    format_err!(
+                        "Error: missing NetworkKeyPairs: peer_id: {}",
+                        self.peer_id.short_str()
+                    )
+                })?
+                .identity_keypair
+                .public_key();
+
+            Ok(bootstrap_address.append_prod_protos(pubkey, handshake_version))
+        } else {
+            Ok(bootstrap_address.append_test_protos(handshake_version))
+        }
+    }
+
+    /// Use this node config to build a `SeedPeersConfig` with this node as the
+    /// single seed peer (using the given `bootstrap_address` instead of the
+    /// configured `self.advertised_address`).
+    pub fn build_seed_peers(&self, bootstrap_address: NetworkAddress) -> Result<SeedPeersConfig> {
```

Comment:
i agree, but it doesn't seem so easy to extract, since `generator` as well as `config-builder`'s `validator_config` and `full_node_config` need this. the `config-builder` does some tricky things with the `bootstrap_address` as the base for the seed peer address